keiko-sql/src/test/kotlin/com/netflix/spinnaker/q/sql/SqlAckQueueTest.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2023 Apple, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.q.sql
+
+import com.netflix.spinnaker.kork.sql.test.SqlTestUtil
+import com.netflix.spinnaker.q.AckAttemptsAttribute
+import com.netflix.spinnaker.q.DeadMessageCallback
+import com.netflix.spinnaker.q.TestMessage
+import com.netflix.spinnaker.q.metrics.EventPublisher
+import com.netflix.spinnaker.time.MutableClock
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.reset
+import org.jetbrains.spek.api.Spek
+import org.jetbrains.spek.api.dsl.describe
+import org.jetbrains.spek.api.dsl.given
+import org.jetbrains.spek.api.dsl.it
+import org.jetbrains.spek.api.dsl.on
+import org.junit.platform.runner.JUnitPlatform
+import org.junit.runner.RunWith
+import java.util.*
+
+
+@RunWith(JUnitPlatform::class)
+class SqlAckQueueTest : Spek({
+  describe("both values of resetAttemptsOnAck") {
+    // map of resetAttemptsOnAck to expected number of ackAttempts still on the message after ack
+    val flagToAckAttempts = mapOf(
+      true to 0,
+      false to 1
+    )
+
+    // check both values of resetAttemptsOnAck
+    flagToAckAttempts.forEach { resetFlag, expectedAckAttempts ->
+      val testDescription = "SqlQueue with resetAttemptsOnAck = $resetFlag"
```

Comment:
I don't see this in the test output (with no local changes).  Here's what I see:
<img width="711" alt="image" src="https://github.com/spinnaker/orca/assets/82477955/4d8ba58d-fe3b-4c57-b9df-f9da8f76be71">